cmd/azure/azure.go

```diff
@@ -88,15 +113,93 @@ func run() error {
 			Hyperthreading: types.HyperthreadingEnabled,
 		},
 		Platform: types.Platform{
-			Azure: &azure.Platform{},
+			Azure: &azure.Platform{
+				Region: ec.Region,
+			},
 		},
 		PullSecret: string(pullSecret),
 	}
+	return &cfg, nil
+}
+
+func run() error {
+	rootCmd := &cobra.Command{
+		Use:  "./azure [component]",
+		Long: "Azure Red Hat OpenShift dispatcher",
+	}
+	rootCmd.PersistentFlags().StringP("loglevel", "l", "Debug", "Valid values are [Debug, Info, Warning, Error]")
+	rootCmd.PersistentFlags().StringP("name", "n", "demo-cluster", "Cluster name value")
+	rootCmd.Printf("gitCommit %s\n", gitCommit)
+	rootCmd.Execute()
 
+	name, err := rootCmd.Flags().GetString("name")
+	if err != nil {
+		return err
+	}
+
+	cfg, err := getInstallConfig(name)
+	if err != nil {
+		return errors.Wrap(err, "failed to get InstallConfig")
+	}
+
+	defaults.SetInstallConfigDefaults(cfg)
+	if err := validation.ValidateInstallConfig(cfg, openstackvalidation.NewValidValuesFetcher()).ToAggregate(); err != nil {
+		return errors.Wrap(err, "invalid install config")
+	}
 	fmt.Println(cfg)
 
+	data, err := yaml.Marshal(cfg)
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal InstallConfig")
+	}
+
+	// doing this to prevent the stdin questions.
+	ic := &installconfig.InstallConfig{}
+	ic.Config = cfg
+	ic.File = &asset.File{
+		Filename: "install-config.yaml",
+		Data:     data,
+	}
+	waIc := []asset.WritableAsset{ic}
+
 	// TODO: Start consuming assets. Assets need to be file system agnostic
 	// Like: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/ssh.go#L49
 
+	// TODO re-implement the store to not use the filesystem?
+	directory := path.Join("clusters", name)
+	assetStore, err := store.NewStore(directory)
+	if err != nil {
+		return errors.Wrap(err, "failed to create asset store")
+	}
+
+	targets := waIc
+	targets = append(targets, targetassets.IgnitionConfigs...)
+	targets = append(targets, targetassets.Manifests...)
+	targets = append(targets, targetassets.Cluster...)
```

Comment:
note: somewhere in the above is the azure clientid and clientSecret, need to find those..